### PR TITLE
feat: florence2 fine-tuning

### PR DESCRIPTION
**Pull request by Dayof**
# Description

Add florence2-ft with the features:
  - Run zero-shot florence2 with large model. 
  - Run zero-shot florence2 with small model. 
  - Run fine-tuned florence2 models for the following tasks: [
            PromptTask.CAPTION_TO_PHRASE_GROUNDING,
            PromptTask.OBJECT_DETECTION,
            PromptTask.CAPTION,
            PromptTask.OCR_WITH_REGION,
        ]
  - Execute prediction with a batch of images (pillow) or a video (numpy).  
  - It contains 2 postprocessing: remove whole image predictions + nms.

A new installation package was added: `florence2-ft`.

# Extras

- Remove files that shouldn't be commit: `.devcontainer.json`
- Restrict python until 3.12.
- Add flash-attn on poetry and remove from the installation script.
- Packages that includes flash-attn: all florence2 + qwen2-vl.
- Use cuda 11.8 with torch + flash-attn. 
- Replace opencv-python to opencv-python-headless.
- Move all images/videos to `shared_data` to avoid duplicated data across tests.
- In the end of the tests share the top 10 slowest tests.
- raise not implemented on .to() for flux1 since it wasn't doing anything, self._pipeline wasn't assigned to any object, also flux1 is always running on CPU when calling this func: `enable_sequential_cpu_offload`
- Skip `tests/tools/test_text_to_image.py` since it's not actually using the flux configs correctly. (cc @hugohonda fix this pls)

# Improve tests

Flux1 tests were too slow (approx. 30 minutes to run in the gpu runner). Some improvements:
- Share the same model (scope=session).
- Generate smaller images for the tests. 

Results:
- until 83 % : ====== 1 failed, 81 passed, 2 skipped, 34 warnings in 4112.58s (1:08:32) =======
- after refac, until 83 %: ====== 1 failed, 80 passed, 2 skipped, 34 warnings in 1501.81s (0:25:01) =======

### CODE REVIEW

File: vision_agent_tools/models/flux1.py [hunk 1]
```diff
@@ -287,6 +188,102 @@ def __call__(
 
         return output.images
 
-    def to(self, device: Device):
-        self._pipeline.to(device=device.value)
-        self._device = device
```
Comment (Dayof) [inline vision_agent_tools/models/flux1.py:292]: this wasn't doing anything, self._pipeline wasn't assigned to any object

File: vision_agent_tools/shared_types.py [hunk 2]
```diff
@@ -124,3 +124,76 @@ class Config:
...
+
+
+class Florence2SegmentationResponse(BaseModel):
+    labels: list[str]
+    polygons: list[list[list[float]]]
+
+
+class Florence2OpenVocabularyResponse(BaseModel):
+    bboxes: list[list[float]]
+    bboxes_labels: list[str]
```
Comment (hrnn) [inline vision_agent_tools/shared_types.py:177]: should we cast it to `labels` to keep consistency ? 
```suggestion
    bboxes_labels: list[str] = Field(alias="labels")
```

Comment (Dayof) [reply vision_agent_tools/shared_types.py:177]: I think is better to keep the way florence2 sends, to not get confused by the polygon labels

**Review by hrnn — APPROVED** [review]
overall it LGTM :shipit:

**Pull request merged**